### PR TITLE
Various benchmarks improvements

### PR DESCRIPTION
**Pull request by apelisse**
- Remove deserialization from benchmarks,
- Add actual pod to test performance,
- Add benchmark for converting objects to TypedValue

### CODE REVIEW

File: merge/pod_test.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+import (
+	"io/ioutil"
+	"testing"
+
+	. "sigs.k8s.io/structured-merge-diff/internal/fixture"
+	"sigs.k8s.io/structured-merge-diff/typed"
+)
+
+var podParser = func() typed.ParseableType {
+	s, err := ioutil.ReadFile("../internal/testdata/k8s-schema.yaml")
```
Comment (lavalamp) [inline merge/pod_test.go]: hrm I thought there was a helper function for reading these?

File: typed/parser_test.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+import (
+	"io/ioutil"
+	"testing"
+
+	yaml "gopkg.in/yaml.v2"
+	"sigs.k8s.io/structured-merge-diff/typed"
+)
+
+func BenchmarkFromUnstructured(b *testing.B) {
+	pod := `
```
Comment (lavalamp) [inline typed/parser_test.go]: Consider reading the pod from the internal data dir instead of copy pasting it multiple places?

**Comment by apelisse** [comment]
updated

**Pull request merged**